react/OrderItems.tsx

```diff
@@ -5,228 +5,448 @@ import React, {
   useContext,
   useMemo,
   useRef,
+  useEffect,
 } from 'react'
 import { useMutation } from 'react-apollo'
-import { updateItems as UpdateItems } from 'vtex.checkout-resources/Mutations'
 import {
-  QueueStatus,
+  updateItems as UpdateItems,
+  addToCart as AddToCart,
+} from 'vtex.checkout-resources/Mutations'
+import { useOrderForm } from 'vtex.order-manager/OrderForm'
+import {
   useOrderQueue,
   useQueueStatus,
+  QueueStatus,
 } from 'vtex.order-manager/OrderQueue'
-import { useOrderForm } from 'vtex.order-manager/OrderForm'
-
-const AVAILABLE = 'available'
-const TASK_CANCELLED = 'TASK_CANCELLED'
 
-enum Totalizers {
-  SUBTOTAL = 'Items',
-  DISCOUNT = 'Discounts',
-}
+import {
+  LocalOrderTaskType,
+  getLocalOrderQueue,
+  popLocalOrderQueue,
+  pushLocalOrderQueue,
+  updateLocalQueueItemIds,
+  UpdateQuantityInput,
+} from './modules/localOrderQueue'
+import {
+  adjustForItemInput,
+  mapItemInputToOrderFormItem,
+  AVAILABLE,
+} from './utils'
 
 interface Context {
+  addItem: (props: Array<Partial<Item>>) => void
   updateQuantity: (props: Partial<Item>) => void
   removeItem: (props: Partial<Item>) => void
 }
 
-interface CancellablePromiseLike<T> extends Promise<T> {
-  cancel: () => void
-}
-
-interface EnqueuedTask {
-  promise?: CancellablePromiseLike<any>
-  variables?: any
+enum Totalizers {
+  SUBTOTAL = 'Items',
+  DISCOUNT = 'Discounts',
 }
 
-const OrderItemsContext = createContext<Context | undefined>(undefined)
-
-const noop = async (_: Partial<Item>) => {}
-
-const maybeUpdateTotalizers = (
-  totalizers: Totalizer[],
-  value: number,
-  oldItem: Item,
+const updateTotalizersAndValue = ({
+  totalizers,
+  currentValue = 0,
+  newItem,
+  oldItem,
+}: {
+  totalizers: Totalizer[]
+  currentValue?: number
   newItem: Item
-) => {
-  if (oldItem.availability !== AVAILABLE) {
-    return {}
+  oldItem?: Item
+}) => {
+  if (oldItem?.availability !== AVAILABLE) {
+    return { totalizers, value: currentValue }
   }
 
-  const oldPrice = oldItem.price * oldItem.quantity
+  const oldPrice = oldItem?.price ?? 0 * (oldItem?.quantity ?? 0)
   const newPrice = newItem.price * newItem.quantity
   const subtotalDifference = newPrice - oldPrice
 
-  const oldDiscount = (oldItem.sellingPrice - oldItem.price) * oldItem.quantity
+  const oldDiscount =
+    (oldItem?.sellingPrice ?? 0 - (oldItem?.price ?? 0)) *
```

Comment:
You also need parenthesis here.
```suggestion
    ((oldItem?.sellingPrice ?? 0) - (oldItem?.price ?? 0)) *
```